src/rAF/update.ts

```diff
@@ -0,0 +1,82 @@
+import { StatefulAnimatingElement } from '../animation';
+import { Entity } from '../forces';
+
+interface UpdateParams<A> {
+  animatingElements: Set<A>;
+  checkReversePlayState: (animatingElement: A) => void;
+  applyForceForStep: (animatingElement: A) => Entity;
+  checkStoppingCondition: (animatingElement: A) => boolean;
+}
+
+/**
+ * The core updater function ran inside requestAnimationFrame.
+ * Iterates through all animating elements in the Set and updates
+ * their local animation state.
+ */
+export function update<A extends StatefulAnimatingElement>({
```

Comment:
`update` is now a generic function that will handle all hooks. Previously, a lot of this logic was duplicated across all hooks, so I pulled it into a single function. Now, calling functions pass in callbacks to handle animation reversal, stopping conditions, and force application.